src/irr/asset/CImageLoaderPNG.cpp

```diff
@@ -152,31 +152,30 @@ asset::IAsset* CImageLoaderPng::loadAsset(io::IReadFile* _file, const asset::IAs
 		Width=w;
 		Height=h;
 	}
-
-	// Convert palette color to true color
-	if (ColorType==PNG_COLOR_TYPE_PALETTE)
+	
+	if (ColorType == PNG_COLOR_TYPE_PALETTE)
 		png_set_palette_to_rgb(png_ptr);
 
 	// Convert low bit colors to 8 bit colors
 	if (BitDepth < 8)
 	{
-		if (ColorType==PNG_COLOR_TYPE_GRAY || ColorType==PNG_COLOR_TYPE_GRAY_ALPHA)
-			png_set_expand_gray_1_2_4_to_8(png_ptr);
-		else
-			png_set_packing(png_ptr);
+		switch (ColorType) {
+			case PNG_COLOR_TYPE_GRAY:
+			case PNG_COLOR_TYPE_GRAY_ALPHA:
+				png_set_expand_gray_1_2_4_to_8(png_ptr);
```

Comment:
don't expand grey or grey-alpha to rgb or rgba

Reply:
It's basically expanding 1/2/4-bit gray to 8-bit gray (if there's any). Grayscale information is still kept intact